File: department-service/src/main/java/github/Jimoou/departmentservice/Entity/Department.java
package github.Jimoou.departmentservice.Entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "departments")
public class Department {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String departmentName;
  private String departmentDescription;
  private String departmentCode;
}

File: react-frontend/src/model/Department.ts
class Department {
  id: number;
  departmentName: string;
  departmentDescription: string;
  departmentCode: string;
  constructor(
    id: number,
    departmentName: string,
    departmentDescription: string,
    departmentCode: string
  ) {
    this.id = id;
    this.departmentName = departmentName;
    this.departmentDescription = departmentDescription;
    this.departmentCode = departmentCode;
  }
}
export default Department;
